standards/cms/src/pkcs7_compat.rs

```diff
@@ -0,0 +1,60 @@
+//! PKCS7 compatibility module.
+//!
+//! The Cryptographic Message Syntax and PKCS7 are mostly the same. The only
+//! difference is that the `EncapsulatedContentInfo` content field has the `Any`
+//! type instead of the `OctetString`. See section 5.2.1. Compatibility with PKCS #7
+//! of RFC5652 for further information.
+use crate::{
+    Any, AuthAttributes, CertificateSet, CmsVersion, ContentType, Digest,
+    DigestAlgorithmIdentifier, DigestAlgorithmIdentifiers, MessageAuthenticationCode,
+    MessageAuthenticationCodeAlgorithm, OriginatorInfo, RecipientInfos, RevocationInfoChoices,
+    SignerInfos, UnauthAttributes,
+};
+use rasn::{AsnType, Decode, Encode};
+
+/// The content is represented in the type EncapsulatedContentInfo
+#[derive(AsnType, Clone, Debug, Decode, Encode, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct EncapsulatedContentInfo {
+    pub content_type: ContentType,
+    pub content: Any,
```

Comment:
Shouldn't this have the following?
```suggestion
    #[rasn(tag(explicit(0))]
    pub content: Any,
```

Reply:
Still seems to work with this change, what does this do?

Reply:
Well it will not be parsed correctly otherwise, as the `Any` type will now contain `EXPLICIT [0]` sequence, so when you try to decode the `Any` type as some other type, it will then fail.

Reply:
looks like that's right according to the RFC I linked, thanks!